pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -159,7 +164,34 @@ func (c *BootstrapTeardownController) removeBootstrap() error {
 		return nil
 	}
 
+	// check to see if bootstrapping is complete
+	bootstrapFinished := false
+	bootstrapFinishedConfigMap, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("bootstrap", metav1.GetOptions{})
+	if err != nil {
+		klog.V(2).Infof("non-fatal error getting configmap: %v", err)
+	} else {
+		if bootstrapFinishedConfigMap.Data["status"] == "complete" {
+			bootstrapFinished = true
+		}
+	}
+	if !bootstrapFinished {
+		// check to see if we have an event yet.
+		_, err := c.kubeClient.CoreV1().Events("kube-system").Get("bootstrap-finished", metav1.GetOptions{})
+		if err != nil {
+			klog.V(2).Infof("non-fatal error getting bootstrap-finished event: %v", err)
+		} else {
+			bootstrapFinished = true
+		}
+	}
+
+	if !bootstrapFinished {
+		c.eventRecorder.Event("DelayingBootstrapTeardown", "cluster-bootstrap is not yet finished")
+		return nil
+	}
+
 	c.eventRecorder.Event("BootstrapTeardownController", "safe to remove bootstrap")
+	// this is ugly until bootkube is updated, but we want to be sure that bootkube has time to be waiting to watch the condition coming back.
+	time.Sleep(10 * time.Second)
 	if err := c.etcdClient.MemberRemove("etcd-bootstrap"); err != nil {
```

Comment:
q: if for some reason the bootstrap vm will be restaretd, will the bootstrap etd will rejoin the cluster automatically?